bulla-contracts/src/functions/BullaFactoring.ts

```diff
@@ -109,3 +110,24 @@ export const createInvoiceReconciledEventV3 = (invoiceId: BigInt, event: Invoice
 
 export const createInvoiceReconciledEventV3_1 = (invoiceId: BigInt, event: InvoicePaidV3_1): InvoiceReconciledEvent =>
   new InvoiceReconciledEvent(getInvoiceReconciledEventId(invoiceId, event));
+
+export const getTargetProtocolFeeFromFundedEvent = (invoiceId: BigInt, pool: User): BigInt => {
+  if (!pool.factoringEvents) {
+    return BigInt.fromI32(0);
+  }
+
+  const invoiceIdStr = invoiceId.toString();
+  const prefix = "InvoiceFunded-" + invoiceIdStr + "-";
+
+  for (let i = 0; i < pool.factoringEvents.length; i++) {
```

Comment:
Change the InvoiceFunded event Id so it is deterministic, i.e. `InvoiceFunded-<invoiceId>-<pool address>`

Then you can load it directly via InvoiceFundedEvent.load